src/System.Management.Automation/engine/Interop/Windows/GetStdHandle.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static unsafe partial class Windows
+    {
+        internal const int STD_INPUT_HANDLE = -10;
+        internal const int STD_OUTPUT_HANDLE = -11;
+        internal const int STD_ERROR_HANDLE = -12;
+
+        [LibraryImport("kernel32.dll", EntryPoint = "GetStdHandle", SetLastError = true)]
+        private static partial nint GetStdHandlePrivate(
+            int nStdHandle);
+
+        internal static SafeFileHandle GetStdHandle(int whichHandle)
```

Comment:
Looking as teh method is used it is not clear that the handle is safe so please rename GetStdHandle to GetSafeStdHandle

Reply:
It is my maintainer's job to monitor the quality of the code from security and maintainability to readability. Unfortunately, it is impossible to formalize all this completely. 
I would like to draw your attention to the fact that the code of this project is studied by many people and is used as a template. The code has a great impact on the community. Therefore, the effort to keep it clean is not only worth it, but it is very expensive because it affects many people's time. Thank you for your understanding.

Reply:
That's totally fine and understandable it's just frustrating to see small incremental progress be blocked by perfection but I'm not a maintainer here so that's just my view. Normally I would probably try and continue forward with the changes but in this specific case it's just not really worth the effort.

Please note I totally understand your perspective and appreciate the work you do here, I just need to weigh up the effort it takes for me to contribute here vs the benefits it brings and in this case it's just not there.